lib/commands/app-management.js

```diff
@@ -55,8 +55,30 @@ commands.queryAppState = async function queryAppState (appId) {
  */
 commands.activateApp = async function activateApp (appId) {
   this.log.debug(`Activating '${appId}'`);
+  const apiLevel = await this.adb.getApiLevel();
+  // Fallback to Monkey in older APIs
```

Comment:
yes, it looks like if a package has multiple activities in MAIN intent then there is no way to determine which of these belongs to which category in Android 6 and older using only adb :(